typescript/web/src/hooks/use-image-pre-fetching.ts

```diff
@@ -63,13 +65,19 @@ export const useImagePrefecthing = () => {
   const previousImageUrl = previousImageData?.image?.url;
   const nextImageUrl = nextImageData?.image?.url;
   useEffect(() => {
-    if (previousImageUrl != null) {
+    if (previousImageUrl != null && canPreFetchPrevious) {
       fetch(previousImageUrl);
     }
-  }, [previousImageUrl]);
+    setCanPreFetchPrevious(false);
+  }, [previousImageUrl, canPreFetchPrevious]);
   useEffect(() => {
-    if (nextImageUrl != null) {
+    if (nextImageUrl != null && canPreFetchNext) {
       fetch(nextImageUrl);
     }
-  }, [nextImageUrl]);
+    setCanPreFetchNext(false);
+  }, [nextImageUrl, canPreFetchNext]);
+  return () => {
+    setCanPreFetchPrevious(true);
+    setCanPreFetchNext(true);
```

Comment:
Can't you just have a single state? Or at-least avoid triggering 2 re-renders?

Reply:
I agree this is not super satisfying but I couldn't find another way of doing it, the issue is simple: you can't manage the preloading of the next and previous image in the same effect as long as the next and previous image URLs are not updated simultaneously - they're in separate queries. I tried to group the queries to obtain them in a single apollo operation but it fails at the borders, so that's what I ended up doing 🤷

Reply:
Aaah, I see how it works now 😉 
But my remark is a pure React problem. You can store `canFetchNext` and `canFetchPrevious` in a single object with:

```tsx
type CanFetchState = {
  canFetchNext: boolean;
  canFetchPrevious: boolean;
}

const useCanFetch = () => useReducer<CanFetchState>(
  (oldState, newState) => ({ ...oldState, ...newState }),
  { canFetchNext: false, canFetchPrevious: false }
);

// ...

const [{ canFetchNext, canFetchPrevious }, setCanFetch] = useCanFetch();
```

Reply:
As `setCanFetch` would accept a partial state as argument, you could then set `canFetchNext`, `canFetchPrevious` or both without triggering 2 state changes 🚀

Reply:
Oh, I see, thanks! 
Just one question about the code above, wouldn't it be the same in terms of re renderings to use `useState< CanFetchState >`  over `useReducer` ?

Reply:
It should but it would force you to always redefine both values at once, even if you only want to set a single one. It's not that important in our case but it makes the code cleaner in general.

Reply:
Ok, I'll apply your proposition 👍